### ISSUE TITLE
"redundant cast" warnings with manifold-ext

### ISSUE THREAD

**Issue by iadcode** [manifold-systems/manifold#270]
Compiling the following code using `manifold-ext` with `-Xlint:all` enabled produces an unexpected warning:
```
public class Foo {
  int i;
  void f() {
    i = 0;
  }
}
```

The warning is:
```
Foo.java:4: warning: [cast] redundant cast to int
    i = 0;
        ^
```

Compiled with AdoptJDK 11.0.8 using:
```
javac -Xlint:all --processor-path manifold-2021.1.8.jar;manifold-rt-2021.1.8.jar;manifold-util-2021.1.8.jar;manifold-ext-2021.1.8.jar;manifold-ext-rt-2021.1.8.jar "-Xplugin:Manifold no-bootstrap" Foo.java
```

It seems to occur for any reassignment of a primitive field/variable to an expression which matches the primitive type. So for example, the following cases do not produce a warning:
```
int i = 0;
```
```
String s;
s = "";
```
```
double d;
d = 0;
```
...though in that last case, the warning would be triggered by `d = 0.0;`.

**Comment by rsmckinney** [comment]
@iadcode thanks for reporting this and for providing all the info!

**Comment by rsmckinney** [comment]
Available in release 2021.1.13
